src/AnalyticsEventReporter.ts

```diff
@@ -0,0 +1,37 @@
+import { AnalyticsEventService } from "./AnalyticsEventService"
+import { AnalyticsConfig } from "./AnalyticsConfig";
+import { EventPayload } from "./EventPayload";
+import { EventAPIResponse } from "./EventAPIResponse";
+
+/** Represents an reporter is responsible for reporting analytics events. */
+export class AnalyticsEventReporter implements AnalyticsEventService {
+
+    private config: AnalyticsConfig;
+    private payload?: EventPayload;
+
+    /**
+     * @param config - necessary analytics config: Must provide one and only
+     * one of API Key or Bearer Token.
+     * 
+     * @param payload - (optional) desired event values to report
+     */
+    constructor(config: AnalyticsConfig, payload?: EventPayload) {
+        const configIsValid: boolean = ('key' in config || 'bearer' in config) 
```

Comment:
I would advise against using `<key> in <object>` because this is true if the value is undefined. `in` just checks if the key exists, and `variable?: <some type>` is equal to `variable?: <some type> | undefined`.